compiler/rustc_resolve/src/late.rs

```diff
@@ -2296,22 +2296,18 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
 
         let prev = self.diagnostic_metadata.current_block_could_be_bare_struct_literal.take();
         if let (true, [Stmt { kind: StmtKind::Expr(expr), .. }]) =
-            (block.could_be_bare_literal, &block.stmts[..])
+            (block.could_be_bare_literal, &block.stmts[..]) && let ExprKind::Type(..) = expr.kind
```

Comment:
```suggestion
            (block.could_be_bare_literal, &block.stmts[..])
            && let ExprKind::Type(..) = expr.kind
```

The [fmt-rfcs discussion is ongoing](https://github.com/rust-dev-tools/fmt-rfcs/issues/169), but as far as I know this should wrap by *either* of the leading candidate formatting rules, because the condition already takes up more than one line.

Reply:
Thank you. I'll fix it.